controller/controller.go

```diff
@@ -38,23 +44,28 @@ func DefaultConfig() Config {
 type Controller interface {
 	// Submit schedules a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to loaded.
-	Submit(req Request) error
+	Submit(req Request) (*task.TaskObject, error)
```

Comment:
No strong feelings. I don't see that much benefit. Looking at fleet this is the same, and there, it is even worse. See https://github.com/giantswarm/app-service/blob/master/service/scheduler-service/fleet_client_scheduler.go#L12-L17